ion/bitstream.go

```diff
@@ -638,7 +639,27 @@ func (b *bitstream) ReadString() (string, error) {
 	b.state = b.stateAfterValue()
 	b.clear()
 
-	return string(bs), nil
+	return toUtf8String(bs, b.pos)
+}
+
+func toUtf8String(buf []byte, pos uint64) (string, error) {
+	n := len(buf)
+	str := strings.Builder{}
+	re := bytes.NewReader(buf)
+	for n > 0 {
```

Comment:
I notice that the `utf8` package offers a function called [`Valid`](https://golang.org/pkg/unicode/utf8/#Valid) that can tell you up-front whether the provided slice contains valid UTF8. Using that will likely be faster than copying a rune at a time to a string builder; if `Valid(buf)` is true, you can return `string(buf)`.

Reply:
Thanks Zack!
It simplified the PR a lot!